File: src/gestorAplicacion/gestion/Operario.java
/* Autores: Joan Sebastian Salazar Montoya, Monica Sofia Restrepo León, 
Luis Alejandro Varela Ojeda, Maria Fernanda Calle Agudelo, Jaider Castañeda Villa
 * FUNCIONALIDAD DE LA CLASE: clase que hereda de Persona, que representa al trabajador 
 * que controla la producción de los productos en las fábricas.
 */
package gestorAplicacion.gestion;
import java.util.ArrayList;
import java.util.List;


import gestorAplicacion.produccion.Fabrica;


public class Operario extends Persona {
	
	/*-------------------Atributos-------------------------*/
	
	private static ArrayList<Meta> metasOperario = new ArrayList<Meta>(List.of(new Meta("Fácil",5,5000),
																				new Meta("Dificil",10,7000)));
	private Fabrica fabrica;
	
	
	/*-------------------Constructores-------------------------*/
	
	
	/*Constructor de la clase operario que recibe todos los parametros*/

	public Operario(String nombre, int edad, int cedula, CuentaBancaria cuentaBancaria,Fabrica fabrica) {
		super(nombre, edad, cedula, cuentaBancaria);
		this.fabrica=fabrica;
	}
	/*Constructor de la clase operario que no recibe parametros*/
	public Operario() {
	}
	

	/*-------------------Metodos-------------------------*/

	/*
     * FUNCIONALIDADES EN LAS QUE ESTÁ INVOLUCRADO: Pago a trabajadores
     * 
     * RECIBE: 
	 * Entero que hace referencia al pago que se va a realizar
	 * 
     * DEVUELVE: 
	 * Nada (vacío)
     * 
     * DESCRIPCIÓN:método sobrescrito de la clase Persona
     */
    @Override
	public void recibirSueldo(int total) {
		fabrica.getCuentaBancaria().descontarFondos(total);
		this.getCuentaBancaria().anadirFondos(total);
	}


	@Override
	public String toString() {
		return "\nNombre: "            + getNombre()            + "\n"
        +      "Edad: "                + getEdad()              + "\n"
        +      "Cedula: "              + getCedula()            + "\n"
		+ 	   "Fabrica: "			   + getFabrica()           + "\n";
	}

	
	/*-------------------Getter y setter-------------------------*/
	
	
	public Fabrica getFabrica() {
		return fabrica;
	}
	public void setFabrica(Fabrica fabrica) {
		this.fabrica = fabrica;
	}
	public static ArrayList<Meta> getMetasOperario(){
		return metasOperario;
	}
	public static void setMetasOperario(ArrayList<Meta> metasOperario){
		Operario.metasOperario = metasOperario;
	}
	
}

File: Python/src/gestorAplicacion/gestion/Operario.py
# /* Autores: Joan Sebastian Salazar Montoya, Monica Sofia Restrepo León, 
# Luis Alejandro Varela Ojeda, Maria Fernanda Calle Agudelo, Jaider Castañeda Villa
#  * FUNCIONALIDAD DE LA CLASE: clase que hereda de Persona, que representa al trabajador 
#  * que controla la producción de los productos en las fábricas.
#  */
from gestorAplicacion.gestion.Meta import Meta
from gestorAplicacion.gestion.Persona import Persona

#aaa
#bbb
#ccc
#ddd

class Operario(Persona):
    #/*-------------------Atributos-------------------------*/
    metasOperario = [
        Meta("Fácil", 5, 5000),
        Meta("Dificil", 10, 7000)
    ]
    listaOperarios = []
    #/*-------------------Constructores-------------------------*/
    def __init__(self, nombre, edad, cedula, cuentaBancaria, fabrica):
        super().__init__(nombre, edad, cedula, cuentaBancaria)
        self.fabrica = fabrica
        self.listaOperarios.append(self)

    #/*-------------------Metodos-------------------------*/
    # 	/*
    #  * FUNCIONALIDADES EN LAS QUE ESTÁ INVOLUCRADO: Pago a trabajadores
    #  * 
    #  * RECIBE: 
	#  * Entero que hace referencia al pago que se va a realizar
	#  * 
    #  * DEVUELVE: 
	#  * Nada (vacío)
    #  * 
    #  * DESCRIPCIÓN:método sobrescrito de la clase Persona
    #  */
    #override
    def recibirSueldo(self, total):
        self.fabrica.getCuentaBancaria().descontarFondos(total)
        self.getCuentaBancaria().anadirFondos(total)

    def __str__(self):
        return (
            "\nNombre: " + self.getNombre() + "\n"
            + "Edad: " + str(self.getEdad()) + "\n"
            + "Cedula: " + str(self.getCedula()) + "\n"
            + "Fabrica: " + str(self.getFabrica()) + "\n"
        )
    #/*-------------------Getter y setter-------------------------*/
    def getFabrica(self):
        return self.fabrica

    def setFabrica(self, fabrica):
        self.fabrica = fabrica

    @classmethod
    def getMetasOperario(cls):
        return cls.metasOperario

    @classmethod
    def setMetasOperario(cls,metasOperario):
        cls.metasOperario = metasOperario

    @classmethod
    def getListaOperarios(cls):
        return cls.listaOperarios

    @classmethod
    def setListaOperarios(cls,listaOperarios):
        cls.listaOperarios = listaOperarios
    